build/build_test.go

```diff
@@ -47,11 +47,13 @@ func TestCollectEnv(t *testing.T) {
 	}
 
 	// Set environment variables
+	defer func() {
+		for key, _ := range env {
+			assert.NoError(t, os.Unsetenv(key))
+		}
+	}()
 	for key, value := range env {
 		assert.NoError(t, os.Setenv(key, value))
-		defer func() {
```

Comment:
defer func(keyToUnset string) {
			assert.NoError(t, os.Unsetenv(keyToUnset))
		}(key)

Would also work too, not?

Reply:
Worth giving it a try @sverdlov93 .